test/puppetlabs/puppetdb/testutils.clj

```diff
@@ -23,45 +23,36 @@
          :prefix "/pdb/cmd"
          :version :v1))
 
-(defn test-db-config
-  "This is a placeholder function; it is supposed to return a map containing
-  the database configuration settings to use during testing.  We expect for
-  it to be overridden by another definition from the test config file, so
-  this implementation simply throws an exception that would indicate that our
-  config file was invalid or not read properly."
+(defn create-hsqldb-map
+  "Returns a database connection map with a reference to a new in memory HyperSQL database"
   []
-  (throw (IllegalStateException.
-          (str "No test database configuration found!  Please make sure that "
-               "your test config file defines a no-arg function named "
-               "'test-db-config'."))))
-
-(defn load-test-config
-  "Loads the test configuration file from the classpath.  First looks for
-  `config/local.clj`, and if that is not found, falls back to
-  `config/default.clj`.
-
-  Returns a map containing the test configuration.  Current keys include:
-
-    :testdb-config-fn : a no-arg function that returns a hash of database
-        settings, suitable for passing to the various `clojure.java.jdbc`
-        functions."
+  {:classname "org.hsqldb.jdbcDriver"
+   :subprotocol "hsqldb"
+   :subname (str "mem:"
+                 (java.util.UUID/randomUUID)
+                 ";hsqldb.tx=mvcc;sql.syntax_pgs=true")})
+
+(defn get-env [var & [default]]
+  (or (System/getenv var) default))
+
+(def create-postgres-map
+  {:classname "org.postgresql.Driver"
+   :subprotocol "postgresql"
+   :subname (get-env "PUPPETDB_DBSUBNAME" "//127.0.0.1:5432/puppetdb_test")
+   :user (get-env "PUPPETDB_DBUSER" "puppetdb")
+   :password (get-env "PUPPETDB_DBPASSWORD" "puppetdb")})
+
+(defn test-db*
   []
-  (binding [*ns* (create-ns 'puppetlabs.puppetdb.testutils)]
-    (try
-      (load "/config/local")
-      (catch java.io.FileNotFoundException ex
-        (load "/config/default")))
-    {:testdb-config-fn test-db-config}))
+  (let [dbtype (get-env "PUPPETDB_DBTYPE" "postgres")]
+    (if (= dbtype "postgres")
+      create-postgres-map
+      (create-hsqldb-map))))
 
 ;; Memoize the loading of the test config file so that we don't have to
 ;; keep going back to disk for it.
-(def test-config
-  (memoize load-test-config))
-
-(defn test-db
-  "Return a map of connection attrs for the test database"
-  []
-  ((:testdb-config-fn (test-config))))
+(def test-db
+  (memoize test-db*))
```

Comment:
Agreed, creating a new DB is easy, easier for sure than cleaning up after yourself. However HSQL won't be around for long, so it sounds like having one is the right thing to do. I suspect we'll be cleaning up after this for a while, i.e. what happens if a failed test happens, leaves some tables around etc.